### PR TITLE
Sign up to app

### PR DESCRIPTION
**Pull request by faraznaeem**
All test are green
Please merge quickly

### CODE REVIEW

**Review by oliverochman — APPROVED** [review]
LGTM!

**Review by vick3d — APPROVED** [review]
awesome work guys!

**Review by amandagibson — APPROVED** [review]
looks great!

**Review by Joberg95 — APPROVED** [review]
Looks good to me!

**Comment by aleciabenjamin** [comment]
You guys kick ass!

**Review by LauraRe — APPROVED** [review]
Looks amazing!

File: features/signup.feature [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Feature: We are signing up
```
Comment (tochman) [inline features/signup.feature:1]: I would rephrase this feature title to:
`A visitor can create an account and become a registered user`
or something like that

File: features/signup.feature [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+As a user
+    In order to get top my mailbox
+    I would like to login to the app
```
Comment (tochman) [inline features/signup.feature:5]: Please revisit the language on this story. Not up to standard. Also, please mind the formatting.

File: features/signup.feature [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+Scenario:
```
Comment (tochman) [inline features/signup.feature:7]: Please add a scenario title (not a blocker though)

File: features/signup.feature [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    Given I am on the landing page
+    And I click on the 'Login' button
+    Then I fill in 'Email' with my 'email@email.con'
```
Comment (tochman) [inline features/signup.feature:10]: I would omit the word `my` (not a blocker though)

File: features/signup.feature (hunk 2 above)
Comment (tochman) [reply features/signup.feature:5]: Please redo the whole thing. This is a feature about signing up to the system, not logging in.

File: features/step_definitions/steps.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
+Given("I am on the landing page") do
+  visit 'welcome/index'
```
Comment (tochman) [inline features/step_definitions/steps.rb:2]: Please user Rails path helpers when navigating using Capybara

File: features/step_definitions/steps.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  Then("I should see {string}") do |string|
+    # Successfully logged in.
+  end
+
+
+
+
+
+
```
Comment (tochman) [inline features/step_definitions/steps.rb:27]: Please avoid empty lines and mind the formatting in the entire file.

File: features/step_definitions/steps.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  
+  Then("I will in {string} with my {string}") do |string, string2|
+    fill_in string, with: 'password'
+
+  end
+
+
+  Then("I should see {string}") do |string|
+    # Successfully logged in.
+  end
```
Comment (tochman) [inline features/step_definitions/steps.rb:21]: Please, remove unused step definitions. Don't leave comments in submitted code.

File: features/step_definitions/steps.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  Given("I click on the {string} button") do |string|
+    click_on(string)
+  end
+  
+  Then("I fill in {string} with my {string}") do |y, x|
+    fill_in y, with: x
+  end
+  
+  Then("I will in {string} with my {string}") do |string, string2|
+    fill_in string, with: 'password'
```
Comment (tochman) [inline features/step_definitions/steps.rb:14]: Please mind the naming of the arguments/variables. Also, IF you set a value to an argument, please use it. Avoid hard-coded values.

File: features/step_definitions/steps.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+Given("I am on the landing page") do
+  visit 'welcome/index'
+end
+  
+  Given("I click on the {string} button") do |string|
+    click_on(string)
+  end
+  
+  Then("I fill in {string} with my {string}") do |y, x|
```
Comment (tochman) [inline features/step_definitions/steps.rb:9]: Single letter argument names? Really???

File: features/step_definitions/steps.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    fill_in y, with: x
+  end
+  
+  Then("I will in {string} with my {string}") do |string, string2|
```
Comment (tochman) [inline features/step_definitions/steps.rb:13]: This looks almost the same as the step definition above (see line 9). Avoid duplications like this.

File: features/step_definitions/steps.rb [hunk 11]
```diff
@@ -0,0 +1,27 @@
+Given("I am on the landing page") do
+  visit 'welcome/index'
+end
+  
+  Given("I click on the {string} button") do |string|
+    click_on(string)
```
Comment (tochman) [inline features/step_definitions/steps.rb:6]: Please make a choice regarding code style. If you use poetry mode (as you do below), please be consistent.

File: features/step_definitions/steps.rb [hunk 12]
```diff
@@ -0,0 +1,27 @@
+Given("I am on the landing page") do
+  visit 'welcome/index'
+end
+  
+  Given("I click on the {string} button") do |string|
```
Comment (tochman) [inline features/step_definitions/steps.rb:5]: Please avoid generic argument/variable names.

**Review by tochman — CHANGES_REQUESTED** [review]
Not up to standard. The title is vague, description is missing. There are no quick merges around here. 😳 Your git flow is not following our best practices and the commit messages are way to ambiguous.

File: features/signup.feature (hunk 4 above)
Comment (tochman) [reply features/signup.feature:10]: Then -> When

File: features/signup.feature [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    Then I will in 'Password' with my 'password'
```
Comment (tochman) [inline features/signup.feature:11]: See comment above

**Comment by tochman** [comment]
Closing this one...
